src/System.Linq.Expressions/tests/DebugViewTests.cs

```diff
@@ -0,0 +1,507 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Xunit;
+
+namespace System.Linq.Expressions.Tests
+{
+    public static class DebugViewTests
+    {
+        private static readonly PropertyInfo s_debugView = typeof(Expression).GetProperty("DebugView", BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Public);
+
+        [Fact]
+        public static void Constant_Null()
+        {
+            Check("null", Expression.Constant(null));
+            Check("null", Expression.Constant(null, typeof(string)));
+            Check("null", Expression.Constant(null, typeof(int?)));
+       }
+
+        [Fact]
+        public static void Constant_String()
+        {
+            Check("\"bar\"", Expression.Constant("bar", typeof(string)));
+        }
+
+        [Fact]
+        public static void Constant_Primitives_Literals()
+        {
+            Check("False", Expression.Constant(false, typeof(bool)));
+            Check("True", Expression.Constant(true, typeof(bool)));
+            Check("\'a\'", Expression.Constant('a', typeof(char)));
+            Check("42", Expression.Constant(42, typeof(int)));
```

Comment:
Yeah, maybe... I had a discussion with @VSadov on exactly this topic a few days ago and we weren't sure how good the test runner is in reporting the exact case that causes failure. I've seen cases where we were feeding `object` parameters of boxed primitives (through an `IEnumerable<object[]>` enumerating these cases), each having value `42` but I couldn't spot the runtime type so it was tricky to find the case rather than searching for a line number in a call stack.

For primitives like this it may work, but there are many other cases where we construct quite complex expression trees with arguments that don't fit in metadata attributes (though I guess we could then use the iterator yielding `object[]` parameter arrays).